### PR TITLE
feat: add renderOverlay

### PR DESCRIPTION
**Pull request by jdockerty**
This adds the `renderOverlay` function in order to wrap the `toOverlay` functionality which returns the resulting object of having the overlay applied.

### CODE REVIEW

File: testdata/kubecfg_test.jsonnet [hunk 1]
```diff
@@ -78,6 +78,25 @@ local result =
...
+  };
+  local expectedOverlayObj = {
+    a: {
+      b: {
+        c: 2,
+        d: 10,
+      },
+    },
+  };
+  std.assertEqual(kubecfg.renderOverlay(testRenderOverlay, import 'overlay.json'), expectedOverlayObj) &&
```
Comment (jdockerty) [inline testdata/kubecfg_test.jsonnet:97]: This doesn't actually work at the moment, I'm trying to figure out why :thinking: 

```
kubecfg show kubecfg_test.jsonnet
ERROR error reading kubecfg_test.jsonnet: RUNTIME ERROR: Field does not exist: renderOverlay
```

**Review by mkmik — COMMENTED** [review]
I don't think a function like that is useful as stdlib

File: testdata/kubecfg_test.jsonnet (hunk 1 above)
Comment (jdockerty) [reply testdata/kubecfg_test.jsonnet:97]: Nevermind, I'm dumb :facepalm: 

I'm using `kubecfg` compiled myself which doesn't have the updated `lib/kubecfg.libsonnet`

**Comment by jdockerty** [comment]
@mkmik Do you think the `toOverlay` should be modified to return the resulting object instead?

I.e. `toOverlay(obj, overlay)`

**Pull request closed without merging**